.github/workflows/enos-run.yml

```diff
@@ -14,25 +14,28 @@ env:
   PKG_NAME: vault
 
 jobs:
+  # Generate Enos scenario matrix based on artifact-name input to test
+  generate-enos-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      enos-scenarios: ${{ steps.enos-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Create Enos scenario matrix
+        id: enos-matrix
+        run: ./.github/scripts/generate-enos-matrix.sh ${{ inputs.artifact-name }}
```

Comment:
This is an interesting idea but I find it a bit wonky to deal with all the when we modify it. Perhaps instead we could have an oss.json and ent.json file that are strictly UTF-8 encoded JSON that define the matrix in an easier to read and modify war. Then, instead of dynamically composing it with a script we could pass the edition from the build job as a variable to determine which file to read. You should be able to read the entire contents of the json file and make it available like:
```yaml
    steps:
    - id: read-matrix
      run: |
        JSON=$(cat ./${{ inputs.edition }}.json)
        echo "::set-output name=matrix::${JSON//'%'/'%25'}"
```